tests/integration_tests/functional/test_mmds.py

```diff
@@ -771,31 +789,23 @@ def test_mmds_snapshot(bin_cloner_path):
     _populate_data_store(test_microvm, data_store)
     ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
 
-    # Ensure that token created on the baseVM is not valid inside the clone.
+    # Mmds V2 is not available with snapshots.
+    # Test that `PUT` requests are not allowed.
     cmd = 'curl -m 2 -s'
-    cmd += ' -X GET'
-    cmd += ' -H  "X-metadata-token: {}"'.format(token)
-    cmd += ' http://169.254.169.254/latest/meta-data/'
+    cmd += ' -X PUT'
+    cmd += ' -H  "X-metadata-token-ttl-seconds: 1"'
+    cmd += ' http://169.254.169.254/latest/api/token'
     _, stdout, stderr = ssh_connection.execute_command(cmd)
-    _assert_out(stdout, stderr, "MMDS token not valid.")
-
-    # Generate new session token.
-    token = _generate_mmds_session_token(
-        ssh_connection,
-        ipv4_address="169.254.169.254",
-        token_ttl=60
-    )
+    expected = "Not allowed HTTP method."
+    _assert_out(stdout, stderr, expected)
 
-    # Ensure the newly created token is valid.
-    cmd = 'curl -m 2 -s'
-    cmd += ' -X GET'
-    cmd += ' -H  "X-metadata-token: {}"'.format(token)
-    cmd += ' http://169.254.169.254/latest/meta-data/'
+    # Fetch metadata using V1 requests.
+    cmd = 'curl -s http://169.254.169.254/latest/meta-data/ami-id/'
     _, stdout, stderr = ssh_connection.execute_command(cmd)
-    _assert_out(stdout, stderr, "ami-id")
+    _assert_out(stdout, stderr, 'ami-12345678')
 
 
-def test_mmds_negative(test_microvm_with_api, network_config):
+def test_mmds_v2_negative(test_microvm_with_api, network_config):
```

Comment:
The default length of a method accepted by pylinter is 100 LOC, maybe we can split this test